src/main/java/uk/gov/hmcts/reform/slc/services/servicebus/MessageProcessor.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.hmcts.reform.slc.services.servicebus;
+
+import com.microsoft.azure.servicebus.IMessage;
+import com.microsoft.azure.servicebus.IMessageReceiver;
+import com.microsoft.azure.servicebus.primitives.ServiceBusException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.slc.services.SendLetterJob;
+
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+@Component
+public class MessageProcessor {
+    private static final Logger logger = LoggerFactory.getLogger(SendLetterJob.class);
+
+    private final Supplier<IMessageReceiver> receiverProvider;
+
+    public MessageProcessor(Supplier<IMessageReceiver> receiverProvider) {
+        this.receiverProvider = receiverProvider;
+    }
+
+    /**
+     * Reads message from a queue and passes is as an argument to action.
+     */
+    public void handle(Function<IMessage, MessageHandlingResult> action) {
+
+        IMessageReceiver messageReceiver;
+
+        try {
+            messageReceiver = receiverProvider.get();
+        } catch (ConnectionException e) {
+            logger.error("Unable to connect to Service Bus", e);
+            return;
+        }
+
+        IMessage message = null;
+        try {
+            message = messageReceiver.receive();
+        } catch (InterruptedException | ServiceBusException e) {
+            logger.error("Unable to read message from queue", e);
+        }
```

Comment:
should we `return` here too? we are catching exception so not like just a normal return of null as a message, which later gets reported as a trace

Reply:
no, we still need to close the connection.